### PR TITLE
Adding FUSE option direct_io

### PR DESCRIPTION
**Pull request by souravgupta-msft**
direct_io can be used to bypass the kernel cache. It disables the use of page cache in the kernel.

TODO: add this option for fuse2.

### CODE REVIEW

File: component/libfuse/libfuse_handler_test.go [hunk 1]
```diff
@@ -58,12 +58,13 @@ func (suite *libfuseTestSuite) TestDefault() {
...
 	suite.assert.False(suite.libfuse.disableWritebackCache)
 	suite.assert.True(suite.libfuse.ignoreOpenFlags)
+	suite.assert.False(suite.libfuse.directIO)
 }
 
 func (suite *libfuseTestSuite) TestConfig() {
 	defer suite.cleanupTest()
 	suite.cleanupTest() // clean up the default libfuse generated
-	config := "allow-other: true\nread-only: true\nlibfuse:\n  attribute-expiration-sec: 60\n  entry-expiration-sec: 60\n  negative-entry-expiration-sec: 60\n  fuse-trace: true\n  disable-writeback-cache: true\n  ignore-open-flags: false\n"
+	config := "allow-other: true\nread-only: true\nlibfuse:\n  attribute-expiration-sec: 60\n  entry-expiration-sec: 60\n  negative-entry-expiration-sec: 60\n  fuse-trace: true\n  disable-writeback-cache: true\n  ignore-open-flags: false\n  direct-io: true\n"
```
Comment (gapra-msft) [inline component/libfuse/libfuse_handler_test.go:67]: how safe is this? Should we add an e2e tests with this enabled?

Comment (souravgupta-msft) [reply component/libfuse/libfuse_handler_test.go:67]: direct_io is disabled by default. If a user wants to disable kernel cache, then they can enable it.
I will add some e2e tests for it.

Comment (souravgupta-msft) [reply component/libfuse/libfuse_handler_test.go:67]: Also this option is supported in v1. So, in my opinion we should support it in v2 as well.

Comment (souravgupta-msft) [reply component/libfuse/libfuse_handler_test.go:67]: Yes

**Review by vibhansa-msft — APPROVED** [review]

File: component/libfuse/libfuse_handler_test.go (hunk 1 above)
Comment (souravgupta-msft) [reply component/libfuse/libfuse_handler_test.go:67]: Added it for fuse2

**Pull request merged**